integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java

```diff
@@ -439,8 +435,42 @@ private int writeInternal(String path, ByteBuffer buf, long size, long offset, F
     final long fd = fi.fh.get();
     CreateFileEntry<FileOutStream> ce = mCreateFileEntries.getFirstByField(ID_INDEX, fd);
     if (ce == null) {
-      LOG.error("Cannot find fd for {} in table", path);
-      return -ErrorCodes.EBADFD();
+      // error out or
+      // if readOrWrite flag detected, close the inputstream,
+      // delete file and create file for overwrite
+      final int flags = fi.flags.get();
+      FileInStream is = mOpenFileEntries.get(fd);
+      if (is == null || !AlluxioFuseUtils.isOpenReadWrite(flags)) {
+        LOG.error("Cannot find fd for {} in table", path);
+        return -ErrorCodes.EBADFD();
+      }
+      if (offset != 0) {
+        LOG.error(String.format("Cannot overwrite file {} with offset {}. "
+            + "File is opened with flags 0x%x", path, offset, fi.flags.get()));
+        return -ErrorCodes.EIO();
+      }
+      try {
+        mReleasingReadEntries.put(fd, is);
+        try {
+          synchronized (is) {
+            is.close();
+          }
+        } finally {
+          mReleasingReadEntries.remove(fd);
+        }
+        final AlluxioURI uri = mPathResolverCache.getUnchecked(path);
+        if (mFileSystem.exists(uri)) {
+          mFileSystem.delete(uri);
+        }
+        // TODO(lu) will multiple threads read()/write() concurrently?
```

Comment:
It is possible.

Reply:
I would suggest testing with 1M client buffer to see if the fuse server side receives two concurrent write.
If there is a problem, that can be handled in another PR.